segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/fs/FSStorageTest.java

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.fs;
+
+import io.pravega.common.io.FileHelpers;
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.storage.SegmentHandle;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageNotPrimaryException;
+import io.pravega.segmentstore.storage.StorageTestBase;
+import io.pravega.test.common.AssertExtensions;
+import lombok.val;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+
+import static io.pravega.test.common.AssertExtensions.assertThrows;
+
+/**
+ * Unit tests for FSStorage.
+ */
+public class FSStorageTest extends StorageTestBase {
+    private File baseDir = null;
+    private FSStorageConfig adapterConfig;
+
+    @Before
+    public void setUp() throws Exception {
+        this.baseDir = Files.createTempDirectory("test_nfs").toFile().getAbsoluteFile();
+        this.adapterConfig = FSStorageConfig
+                .builder()
+                .with(FSStorageConfig.ROOT, this.baseDir.getAbsolutePath())
+                .build();
+    }
+
+    @After
+    public void tearDown() {
+            FileHelpers.deleteFileOrDirectory(baseDir);
+            baseDir = null;
+    }
+
+    //region Fencing tests
+
+    /**
+     * Tests fencing abilities. We create two different Storage objects with different owner ids.
+     * Part 1: Creation:
+     * * We create the Segment on Storage1:
+     * ** We verify that Storage1 can execute all operations.
+     * ** We verify that Storage2 can execute only read-only operations.
+     * * We open the Segment on Storage2:
+     * ** We verify that Storage1 can execute only read-only operations.
+     * ** We verify that Storage2 can execute all operations.
+     */
+    @Override
```

Comment:
This is missing the `@Test` annotation, I don't think this is running without it.